File: examples/hosted-wallet-kt/pfi/src/main/kotlin/api/Offerings.kt
package api

import spark.Response
import spark.Spark.get
import tbdex.sdk.http.GetOfferingsResponseBody

class Offerings(private val offeringsRepository: data.Offerings) {
    init {
        get("/offerings") { _, res -> getOfferings(res)}
    }

    private fun getOfferings(res: Response): String {
        println("GET /offerings")

        val offerings = offeringsRepository.getOfferings()
        val responseBody = GetOfferingsResponseBody(offerings)

        return responseBody.toJsonString()
    }
}

File: examples/hosted-wallet-ts/src/pfi/api/offerings.ts
import { Router } from 'express';
import { GetOfferingsResponseBody } from 'tbdex';
import { OfferingRepository } from '../data/offering-repository';

export default function offeringsRouter(offeringsRepository: OfferingRepository) {
  const router = Router();

  router.get('/', (req, res) => {
    console.log('GET /offerings');

    const offerings = offeringsRepository.getOfferings();
    const responseBody = new GetOfferingsResponseBody(offerings);

    res.type('application/json');
    res.send(responseBody.toJSONString());
  });

  return router;
}